src/services/ResultsAggregator.js

```diff
@@ -0,0 +1,82 @@
+const ColorConsole = require('./ColorConsole');
+// const ConfigService = require('./services/Config/ConfigService');
+// TODO use ConfigService getTestRailStatusSkipped, getTestRailStatusFailed, 
+//    and getTestRailStatusPassed Instead of it 2, 5, 1
+
+const RESULT_STATUS_FAILED = 5;
+const RESULT_STATUS_SKIP = 2;
+const RESULT_STATUS_PASS = 1;
+
+
+class ResultsAggregator {
+
+
+  /**
+   *
+   * @param {cytrPostData}
+   * @returns {*}
+   */
+  aggregateDuplicateResults(cytrPostData) {
+    // Replace multiple occurrences of test cases with one output summarizing case, aggregation rules are;
+    // Fail trumps all, anything with a failed leads to the first failed item is the one emitted
+    // If no mixed outcome, the same result is passed through
+    // Pass trumps skip
+    // If all test pass. The last comment is the one emitted.
+    // if all pass or skip modify the comment statement with the prefix of `Summarization of {n} Cypress tests` 
+    const emitPostData = []
+    const mergedResults = {} // this is a map of case_id to merged result object
+
+    cytrPostData.forEach((postData) => {
+        const { case_id, status_id, comment, elapsed, screenshotPaths } = postData;
+        if (Object.hasOwnProperty.call(mergedResults, case_id)) {
+            // If one case failed don't process additional cases, Fail trumps all
+            if (mergedResults[case_id].status_id !== RESULT_STATUS_FAILED) { 
+                // If the current case is a skip then don't processing it
+                if (status_id !== RESULT_STATUS_SKIP) { 
+                    mergedResults[case_id].status_id = status_id;
+                    mergedResults[case_id].comment = comment;
+                    mergedResults[case_id].elapsed = elapsed;
+                    mergedResults[case_id].screenshotPaths = screenshotPaths;
+                    if (status_id === RESULT_STATUS_PASS) {
+                        mergedResults[case_id].countP += 1;
+                    } else {
+                        mergedResults[case_id].countF += 1;
+                    }
+                    mergedResults[case_id].countS += 1;
+                }
+            }
```

Comment:
This is going to stop counting once you've found a failed test.  Also you're incrementing `countS` inside the "status_id *not* RESULT_STATUS_SKIP" if block.

Put your counting logic above line 33:
```suggestion
            const result = mergedResults[case_id];
            switch ( result.status_id ) {
              case RESULT_STATUS_FAILED:
                result.countF++;
                break;
              case RESULT_STATUS_PASS:
                result.countP++;
                break;
              case RESULT_STATUS_SKIP:
                result.countS++;
                break;
            }
            if (mergedResults[case_id].status_id !== RESULT_STATUS_FAILED) { 
                // If the current case is a skip then don't processing it
                if (status_id !== RESULT_STATUS_SKIP) { 
                    mergedResults[case_id].status_id = status_id;
                    mergedResults[case_id].comment = comment;
                    mergedResults[case_id].elapsed = elapsed;
                    mergedResults[case_id].screenshotPaths = screenshotPaths;
                }
            }
```